torch/utils/data/datapipes/iter/callable.py

```diff
@@ -109,13 +109,11 @@ def __getstate__(self):
         if IterDataPipe.getstate_hook is not None:
             return IterDataPipe.getstate_hook(self)
 
-        if DILL_AVAILABLE:
-            dill_function = dill.dumps(self.fn)
-        else:
-            dill_function = self.fn
+        serialized_fn, method = serialize_fn(self.fn, DILL_AVAILABLE)
         state = (
             self.datapipe,
-            dill_function,
+            serialized_fn,
+            method,
```

Comment:
Could we combine these two variables into a tuple for serialization?

Reply:
Done.